src/backend/linalg_internal_gpu/cuTensordot_internal.cu

```diff
@@ -90,175 +101,142 @@ namespace cytnx {
-      cudaDataType_t type = CUDA_C_32F;
-      cutensorComputeType_t typeCompute = CUTENSOR_COMPUTE_32F;
+      cutensorDataType_t type = CUTENSOR_C_32F;
+      cutensorComputeDescriptor_t descCompute = CUTENSOR_COMPUTE_DESC_32F;
 
       hostType alpha = {1.f, 0.f};
       hostType beta = {0.f, 0.f};
 
-      _cuTensordot_internal(out, Lin, Rin, idxl, idxr, type, typeCompute, &alpha, &beta);
+      _cuTensordot_internal(out, Lin, Rin, idxl, idxr, type, descCompute, &alpha, &beta);
     }
 
     void cuTensordot_internal_d(Tensor &out, const Tensor &Lin, const Tensor &Rin,
                                 const std::vector<cytnx_uint64> &idxl,
                                 const std::vector<cytnx_uint64> &idxr) {
       typedef double hostType;
-      cudaDataType_t type = CUDA_R_64F;
-      cutensorComputeType_t typeCompute = CUTENSOR_COMPUTE_64F;
+      cutensorDataType_t type = CUTENSOR_R_64F;
+      cutensorComputeDescriptor_t descCompute = CUTENSOR_COMPUTE_DESC_64F;
 
       hostType alpha = 1.f;
       hostType beta = 0.f;
 
-      _cuTensordot_internal(out, Lin, Rin, idxl, idxr, type, typeCompute, &alpha, &beta);
+      _cuTensordot_internal(out, Lin, Rin, idxl, idxr, type, descCompute, &alpha, &beta);
     }
 
     void cuTensordot_internal_f(Tensor &out, const Tensor &Lin, const Tensor &Rin,
                                 const std::vector<cytnx_uint64> &idxl,
                                 const std::vector<cytnx_uint64> &idxr) {
       typedef float hostType;
-      cudaDataType_t type = CUDA_R_32F;
-      cutensorComputeType_t typeCompute = CUTENSOR_COMPUTE_32F;
+      cutensorDataType_t type = CUTENSOR_R_32F;
+      cutensorComputeDescriptor_t descCompute = CUTENSOR_COMPUTE_DESC_32F;
 
       hostType alpha = 1.f;
       hostType beta = 0.f;
 
-      _cuTensordot_internal(out, Lin, Rin, idxl, idxr, type, typeCompute, &alpha, &beta);
-    }
-
-    void cuTensordot_internal_u32(Tensor &out, const Tensor &Lin, const Tensor &Rin,
-                                  const std::vector<cytnx_uint64> &idxl,
-                                  const std::vector<cytnx_uint64> &idxr) {
-      typedef cytnx_uint32 hostType;
-      cudaDataType_t type = CUDA_R_32U;
-
-      cutensorComputeType_t typeCompute = CUTENSOR_COMPUTE_32U;
-
-      hostType alpha = 1;
-      hostType beta = 0;
-
-      _cuTensordot_internal(out, Lin, Rin, idxl, idxr, type, typeCompute, &alpha, &beta);
-    }
-
-    void cuTensordot_internal_i32(Tensor &out, const Tensor &Lin, const Tensor &Rin,
-                                  const std::vector<cytnx_uint64> &idxl,
-                                  const std::vector<cytnx_uint64> &idxr) {
-      typedef cytnx_int32 hostType;
-      cudaDataType_t type = CUDA_R_32I;
-
-      cutensorComputeType_t typeCompute = CUTENSOR_COMPUTE_32I;
-
-      hostType alpha = 1;
-      hostType beta = 0;
-
-      _cuTensordot_internal(out, Lin, Rin, idxl, idxr, type, typeCompute, &alpha, &beta);
```

Comment:
does u32 and i32 got removed?

Reply:
`cutensorContraction()` has only supported floating-point types at least
since version 1.6.2, so use the default implementation (CPU version). The if-else branch is updated at src/linalg/Tensordot.cpp:101